models/weight_recorder.h

```diff
@@ -131,7 +131,19 @@ class weight_recorder : public Node
   bool user_set_precise_times_;
   bool has_proxies_;
   bool local_receiver_;
-  omp_lock_t writelock;
+
+  struct Parameters_
+  {
+    std::set< long > pre_;
+    std::set< long > post_;
```

Comment:
I agree, but actually I had an issue with Travis and `std::set`. So I picked up your earlier idea and changed these parameters to sorted `vector` and use a binary search to access elements. Fortunately, these two fields will only be set once (or a few times) in the build phase, so the `vector` has not to be resorted during the simulation.

Overall, I think this solution is better than a `set` as the complexity to access elements is also O(log N) and I think a `vector` needs less memory.